msdm/core/problemclasses/mdp/canonicalmdp.py

```diff
@@ -0,0 +1,58 @@
+from msdm.core.problemclasses.mdp.mdp import MarkovDecisionProcess
+from msdm.core.problemclasses.mdp.tabularmdp import TabularMarkovDecisionProcess
+from msdm.core.distributions import DictDistribution
+
+class CanonicalMDP(MarkovDecisionProcess):
+    '''
+    CanonicalMDP is a wrapper class of MDP instances that appropriately implements
+    is_terminal(state) and actions(state) in terms of the simplest primitives
+    typically used to describe MDPs (i.e. just the reward and transition functions).
+
+    is_terminal(state) is a predicate used to denote absorbing states. Following
+    conventions in the literature, an absorbing state transitions to itself regardless
```

Comment:
We should find a source for this

Reply:
Here's a relevant snippet from the [2nd edition of Sutton & Barto](http://incompleteideas.net/book/RLbook2020.pdf), page 57

<img width="571" alt="Screen Shot 2021-11-04 at 10 43 09 AM" src="https://user-images.githubusercontent.com/1276728/140335770-4b32cd4e-75b5-41d3-9215-c1e5fcd02195.png">